dashboard/app/models/script.rb

```diff
@@ -190,23 +195,32 @@ def has_lesson_plan?
 
   def self.setup(default_files, custom_files)
     transaction do
+      scripts = []
       # Load default scripts from yml (csv embedded)
-      default_scripts = default_files.map { |yml| load_yaml(yml, SCRIPT_MAP) }
+      default_files.map { |yml| load_yaml(yml, SCRIPT_MAP) }
       .sort_by { |options, _| options['id'] }
-      .map { |options, data| add_script(options, data) }
+      .map { |options, data| scripts << [options, data]}
 
       custom_i18n = {}
       # Load custom scripts from Script DSL format
-      custom_scripts = custom_files.map do |script|
+      custom_files.map do |script|
+        name = File.basename(script, '.script')
         script_data, i18n = ScriptDSL.parse_file(script)
         stages = script_data[:stages]
         custom_i18n.deep_merge!(i18n)
-        add_script({name: File.basename(script, '.script'),
-                    trophies: false,
-                    hidden: script_data[:hidden].nil? ? true : script_data[:hidden]},
-                   stages.map{|stage| stage[:levels]}.flatten)
+        scripts << [{
+          name: name,
+          trophies: script_data[:trophies],
+          hidden: script_data[:hidden].nil? ? true : script_data[:hidden],
+          id: LEGACY_SCRIPT_ID[name],
```

Comment:
can you just specify this in the .script file (ie. add it to the dsl) so the code doesn't need to know about this anymore?